cvmfs/options.h

```diff
@@ -15,24 +15,57 @@
 namespace CVMFS_NAMESPACE_GUARD {
 #endif
 
-namespace options {
+struct ConfigValue {
+  std::string value;
+  std::string source;
+};
+
+class OptionsManager {
+
+protected:
+  bool HasConfigRepository(const std::string &fqrn, std::string *config_path);
+
+public:
+  OptionsManager() :
+    config_(std::map<std::string, ConfigValue>()) {}
+  virtual ~OptionsManager() {}
+
+  virtual void ParsePath(const std::string &config_file, const bool external) = 0;
+  void ParseDefault(const std::string &fqrn);
+  void ClearConfig();
+  bool IsDefined(const std::string &key);
+  bool GetValue(const std::string &key, std::string *value);
+  bool GetSource(const std::string &key, std::string *value);
+  bool IsOn(const std::string &param_value);
+  std::vector<std::string> GetAllKeys();
+  std::string Dump();
+
+  bool ParseUIntMap(const std::string &path, std::map<uint64_t, uint64_t> *map);
+
+protected:
+  std::map<std::string, ConfigValue> config_;
+
+};  // class OptionManager
+
+
+
+class FastOptionsManager : public OptionsManager {
```

Comment:
It would be good to add a Doxygen comment to this class that explains why it is needed and how it is different from the BashOptionsParser.